File: 59 螺旋矩阵2(模拟)/myCode.py
class Solution:
    def generateMatrix(self, n: int) -> List[List[int]]:
        target = n * n
        num = 1
        levelBegin = 0
        levelEnd = n-1
        verticalBegin = 0
        verticalEnd = n-1
        result = [[0 for i in range(n)] for j in range(n)]
        while num <= target and verticalBegin < verticalEnd:
            # print(verticalBegin, verticalEnd, levelBegin, levelEnd)
            i = verticalBegin
            j = levelBegin
            while j < levelEnd:
                result[i][j] = num
                j += 1
                num += 1
            levelEnd -= 1
            while i < verticalEnd:
                result[i][j] = num
                i += 1
                num += 1
            verticalEnd -= 1
            while j > levelBegin:
                result[i][j] = num
                j -= 1
                num += 1
            levelBegin += 1
            while i > verticalBegin:
                result[i][j] = num
                i -= 1
                num += 1
            verticalBegin += 1
        if verticalBegin == verticalEnd:
            result[verticalBegin][levelBegin] = target
        return result

File: 343 整数拆分(动态规划)/myCode.cpp
class Solution {
public:
    int integerBreak(int n) {
        if(n == 2) return 1;
        if(n == 3) return 2;
        int *num = new int[n];
        for(int i = 0;i < n;i++) num[i] = 0;
        num[1] = 2; num[2] = 3;
        for(int i = 3;i < n;i++){
            num[i] = max(num[i-2] * 2, num[i-3] * 3); 
        }
        return num[n-1];
    }
};